glean-core/rlb/src/private/dynamic/counter.rs

```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use glean_core::{ErrorType, Glean};
+use inherent::inherent;
+
+use crate::{dispatcher, new_metric};
+
+/// Counter metric wrapper around the FFI implementation
+#[derive(Clone)]
+pub struct CounterMetric(pub(crate) u64);
+
+impl CounterMetric {
+    /// The public constructor used by automatically generated metrics.
+    pub fn new(meta: glean_core::CommonMetricData) -> Self {
+        let metric = new_metric!(glean_new_counter_metric, meta);
+        Self(metric)
+    }
+
+    /// Internal only, synchronous API for incremeting a counter
+    pub(crate) fn add_sync(&self, glean: &Glean, amount: i32) {
+        let id = self.0;
+        crate::sys::with_glean(|glean| unsafe { glean.glean_counter_add(id, amount) })
+    }
+}
+
+impl Default for CounterMetric {
```

Comment:
Took me a while to notice that we only need Default because of `new_metric!`. Could use a comment (or figuring out some way to generate the whole type by macro, maybe. In the future, of course.).

Reply:
Yeah, this was a last minute fix for the labeled counters.